src/handlers/transactions.ts

```diff
@@ -118,6 +123,20 @@ export const showTransactionDetailsSheet = (
   if (!network) {
     return;
   }
+
+  const parentTxHash = hash?.includes('-') ? hash.split('-')[0] : hash;
+  const data = metadataStorage.getString(parentTxHash?.toLowerCase() ?? '');
+  const wrappedMeta = data ? JSON.parse(data) : {};
+  let parsedMeta: undefined | SwapMetadata;
+  if (wrappedMeta?.type === 'swap') {
+    parsedMeta = wrappedMeta.data as SwapMetadata;
+  }
+
+  const isReadOnly =
```

Comment:
i believe you could use `canBeResubmitted` instead, so we don't getState here and have the same behavior than speed ups and cancel tx